# MaxFuse: Cross-modality matching of single cells via iterative fuzzy smoothed embedding

<img src="https://github.com/shuxiaoc/maxfuse/tree/main/media" width="100" height="100">

## Description

MaxFuse is a Python package for integrating multi-modal single-cell data under low signal-to-noise ratio regimes. For current single cell cross modality integation methods, the feasibility of cross-modal integration relies on the existence of highly correlated, a priori 'linked features'.  When such linked features are few or uninformative, a scenario that we call 'weak linkage', existing methods fail.  We developed MaxFuse, a cross-modal data integration method that, through iterative co-embedding, data smoothing, and cell matching, leverages all information in each modality to obtain high-quality integration. A prototypical example of weak linkage is the integration of **spatial proteomic data** with **single-cell sequencing data**. For detials, please refer to the [paper](https://www.biorxiv.org/content/10.1101/2023.01.12.523851v2).

This work has been lead by Shuxiao Chen from [Zongming Lab](http://www-stat.wharton.upenn.edu/~zongming/) @Upenn and Bokai Zhu from [Nolan lab](https://web.stanford.edu/group/nolan/) @Stanford.

<img src="https://github.com/shuxiaoc/maxfuse/blob/main/media/fig1.png" width="800" height="280">

## Installation

installation instruction here.

## Usage

linke to sphinx ?

## Code arcive

The analysis presented in the [manuscript](https://www.biorxiv.org/content/10.1101/2023.01.12.523851v2) was also deposited in this GitHub repository, under this [folder](https://github.com/shuxiaoc/maxfuse/tree/main/Archieve). Note in the manuscript we used a development version of ```MaxFuse``` with slightly different grammar and can also be found there. If you require additional inforamtion on the analysis/data, please contact Zongming Ma (zongming@wharton.upenn.edu).

## License

```MaxFuse``` is under the [Academic Software License Agreement](https://github.com/shuxiaoc/maxfuse/blob/main/LICENSE), please use accordingly.

